src/components/color_picker/color_picker.ts

```diff
@@ -209,6 +210,11 @@ export class ColorPicker extends Component<ColorPickerProps, SpreadsheetChildEnv
     },
   });
 
+  get dropdownStyle() {
+    const { height } = this.env.model.getters.getSheetViewDimension();
```

Comment:
Yeah this is false, there is also color pickers in the side panels, in in this case this will be totally wrong. 
Maxheight/style should likely be given as a props.

Reply:
While solving this, I referred google sheet behaviour and update according to that. And about props, that's not updating with changing the screen size.

Reply:
Your props `element` is still not what I had in mind. The `ColorPicker` should have no knowledge of what is its parent. It cannot compute its maxHeight, because this would means that the `ColorPicker` knows 1) it  is in the topBar and 2) knows that the topBar is above the grid, and that a dropdown in the topbar should have maximum the size of the grid.

The `maxHeight` should be computed in the topbar, and given as props in the `ColorPicker`. That way the color picker is totally agnostic of its place in the DOM. All it knows is that it have a props `maxHeight` and should render accordingly.

Reply:
But the problem is in props, it won't update with changing screen size and because of that `maxHeight` will not get properly. And another point is that, Is there any way to check whether side panel is open or not?

Reply:
Did you look at my branch 16.0-fix-responsive-dropdown-menu-fase-test-adrm ? I gave the maxHeight as a props and it's working.

We did have problems with re-sizing the grid not triggering a re-render, but I don't think this is the problem you're having. There is 2 possible things that can happen : 
1) resizing the window cause a render. In this case the topbar is re-rendered, its t-att-style are computed, and the props given to colopicker is changer. Since its props changed, the color picker will re-render with the new props.
2) resizing the window don't cause a render. In this case, neither the t-att-style not the props will be recomputed. 

I'm 90% sure that its impossible for the change you made in the t-att-style to work, but at the same time the props to not be updated.

For your other question, I'm pretty sure it's not possible to know if the side panel is open, but opening/closing the side panel changes the grid dimension so you shouldn't need to worry about the side panel.